bodo/libs/streaming/_groupby.cpp

```diff
@@ -2712,9 +2714,7 @@ GroupbyState::GroupbyState(
     // safely set skip_na_data to true for all SQL aggregations. There is an
     // issue to fix this behavior so that use_sql_rules trumps the value of
     // skip_na_data: https://bodo.atlassian.net/browse/BSE-841
-    bool skip_na_data = true;
-    bool use_sql_rules = true;
-
+    bool skip_na_data = use_sql_rules;
```

Comment:
Looks like `skip_na_data` should be true for Pandas as well. We set it to true in JIT path here: https://github.com/bodo-ai/Bodo/blob/302e4f06f4f6ff3746f1f113fcae83ab2dc6e6dc/bodo/ir/aggregate.py#L522

Here is Pandas example code skipping NAs:
```python
   ...: df = pd.DataFrame(
   ...:     {
   ...:         "A": pd.array([pd.NA, 2, 3, 7] * 3, "Int64"),
   ...:         "B": ["A1", "B1", "C1", "Abc"] * 3,
   ...:         "C": pd.array([6, pd.NA, 4, 4, 1, 3] * 2, "Int64"),
   ...:     }
   ...: )
In [10]: df.groupby("A")["C"].sum()
Out[10]:
A
2     7
3    11
7     7
Name: C, dtype: Int64
```

Reply:
What do you mean? `dropna=True` is on by default but NA data are different rows than keys here. Not sure why `skip_na_data` doesn't do anything.